src/service_daemon.rs

```diff
@@ -1398,13 +1398,16 @@ impl Zeroconf {
             }
         }
 
-        let mut subnet_set: HashSet<u128> = HashSet::new();
+        let mut intf_subnet_set: HashSet<(String, u128)> = HashSet::new();
```

Comment:
one could simplify this, and use the interface index instead of the name to track interface uniqness.

Reply:
But on the other hand, the interface index is an option and therefore is not reliable